flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergSink.java

```diff
@@ -663,9 +663,13 @@ IcebergSink build() {
           }
         }
       }
-
       FlinkMaintenanceConfig flinkMaintenanceConfig =
           new FlinkMaintenanceConfig(table, writeOptions, readableConfig);
+
+      if (flinkWriteConf.uidSuffix() != null) {
+        uidSuffix = flinkWriteConf.uidSuffix();
+      }
```

Comment:
Maybe put the uidSuffix into the `writeOptions` map, and handle the same way as any other config?